Standardize tool error logging

akkulg opened the pull request:
- Standardize tool error logging in order to create error dashboard for tool errors 
- Initially there was no such pattern to filter the tool errors 
- Added Tool Error tag which will help to filter tool errors

<!-- This is an auto-generated comment: release notes by coderabbit.ai -->

## Summary by CodeRabbit

* **New Features**
  * Added configurable speech-to-text prompt to improve transcription of Indian languages, numbers (lakhs/crores), proper names, and technical terms.

* **Refactor**
  * Standardized error messages across voice, analytics, internet search, and system tools with consistent “Tool Error: [function]” prefixes and clearer context.

* **Documentation**
  * Updated environment configuration example to include the new speech-to-text prompt variable.

* **Chores**
  * Improved logging consistency for easier troubleshooting without changing tool behaviors or APIs.

<!-- end of auto-generated comment: release notes by coderabbit.ai -->

akkulg commented on app/agents/voice/automatic/tools/shopify_buddy_test/get_order.py:
```diff
@@ -131,8 +131,8 @@ async def get_shopify_orders(params: FunctionCallParams):
-            await params.result_callback({"error": f"Shopify API error: {e.response.status_code}", "details": e.response.text})
+            logger.error(f"Tool Error: [get_shopify_orders] HTTP error calling Shopify API: {e.response.status_code} - {e.response.text}")
+            await params.result_callback({"error": f"Tool Error: [get_shopify_orders] Shopify API error: {e.response.status_code}", "details": e.response.text})
         except Exception as e:
```
not required

akkulg commented on app/agents/voice/automatic/tools/juspay/analytics.py:
```diff
@@ -660,8 +658,7 @@ async def merchant_offer_analytics(params: FunctionCallParams):
-            f"Critical error in merchant_offer_analytics: {e}", exc_info=True)
+        logger.error(f"Tool Error: [merchant_offer_analytics] Critical error: {e}", exc_info=True)
         await params.result_callback({"error": f"A critical error occurred in the tool function: {e}"})
 
```
not required

akkulg commented on app/agents/voice/automatic/services/charts/chart_tools.py:
```diff
@@ -295,8 +295,8 @@ async def generate_donut_chart(params) -> None:
     except Exception as e:
         error_message = str(e)
-        logger.error(f"[{session_id}] Error generating donut chart: {error_message}")
-        await params.result_callback(f"Error generating chart: {error_message}")
+        logger.error(f"Tool Error: [generate_donut_chart] [{session_id}] Error generating chart: {error_message}")
+        await params.result_callback(f"Tool Error: [generate_donut_chart] Error generating chart: {error_message}")
 
```
not required

akkulg commented on app/agents/voice/automatic/services/charts/chart_tools.py:
```diff
@@ -213,7 +213,7 @@ async def generate_donut_chart(params) -> None:
         # Validate data_type parameter
         valid_data_types = ['currency', 'numericalValue', 'percentage', 'unknown']
         if data_type not in valid_data_types:
-            await params.result_callback(f"Error: Invalid data_type. Must be one of: {valid_data_types}")
+            await params.result_callback(f"Tool Error: [generate_donut_chart] Invalid data_type. Must be one of: {valid_data_types}")
             return
```
not required

akkulg commented on app/agents/voice/automatic/services/charts/chart_tools.py:
```diff
@@ -193,8 +193,8 @@ async def generate_line_chart(params) -> None:
-        await params.result_callback(f"Error generating chart: {error_message}")
+        logger.error(f"Tool Error: [generate_line_chart] [{session_id}] Error generating chart: {error_message}")
+        await params.result_callback(f"Tool Error: [generate_line_chart] Error generating chart: {error_message}")
 
```
not required

akkulg commented on app/agents/voice/automatic/services/charts/chart_tools.py:
```diff
@@ -67,7 +67,7 @@ async def generate_bar_chart(params) -> None:
         if len(series_data) > 1:
-            await params.result_callback("Error: Only one series allowed for bar charts")
+            await params.result_callback("Tool Error: [generate_bar_chart] Only one series allowed for bar charts")
             return
```
not required

swaroopvarma1 commented on app/agents/voice/automatic/tools/system/utils.py:
```diff
@@ -12,7 +13,8 @@ async def get_current_time(params: FunctionCallParams):
         await params.result_callback({"time": current_time})
     except Exception as e:
-        await params.result_callback({"error": str(e)})
+        logger.error(f"Tool Error: [get_current_time] Error getting current time: {e}")
```
logger Exception?
Why logger Error?

badri-singhal commented:
Rebase your branch with release

akkulg committed:
- standardize tool error logging in order to create error dashboard for tool errors
- Initially there was no such pattern to filter the tool errors
- Added Tool Error tag which will help to filter tool errors
